### PR TITLE
Dit advisers tab

### PR DESCRIPTION
**Pull request by retex73**
Trello: https://trello.com/c/BnBgYsuY

Show advisers tab for companies on the One List. 
Advisers tab should show the core team and the global account manager
screenshot 2018-11-26 at 16 42 12

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -44,3 +44,6 @@ selenium-debug.log
 test/acceptance/screenshots/
 cucumber
 seleniumLogs
+
```
Comment (yeahfro) [inline .gitignore]: It doesn't feel like this matches up to the commit message and would have been better as a separate commit

File: src/apps/companies/views/_advisers-table.njk [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{% macro adviserTable(team, labels, ariaLabel) %}
```
Comment (yeahfro) [inline src/apps/companies/views/_advisers-table.njk]: I was expecting the implementation to be a bit more like [Client relationship management in investment projects](https://github.com/uktrade/data-hub-frontend/blob/develop/src/apps/investment-projects/views/team/details.njk#L6).

File: src/apps/companies/transformers/one-list-core-team-to-collection.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const isAccountManager = adviser => adviser.is_global_account_manager
+const isAdviser = adviser => !adviser.is_global_account_manager
+
+module.exports = function transformOneListCoreTeamToCollection (advisers) {
+  const flattenTeam = (team) => {
+    const records = []
```
Comment (yeahfro) [inline src/apps/companies/transformers/one-list-core-team-to-collection.js]: This could definitely be simplified with `map`

File: src/apps/companies/controllers/advisers.js [hunk 4]
```diff
@@ -10,16 +11,15 @@ async function renderAdvisers (req, res, next) {
   try {
     const { name: companyName, id: companyId } = res.locals.company
     const token = req.session.token
-
     const coreTeam = await getCoreTeam(token, companyId)
```
Comment (yeahfro) [inline src/apps/companies/controllers/advisers.js]: This can now use `getOneListGroupCoreTeam` and `getCoreTeam` can be made redundant

File: src/apps/companies/middleware/local-navigation.js [hunk 5]
```diff
@@ -10,11 +10,12 @@ function setCompaniesLocalNav (req, res, next) {
 
   const headquarterType = get(company, 'headquarter_type.name')
   const companyNumber = get(company, 'company_number')
+  const isOneListGroupTier = !isNull(company.one_list_group_tier)
 
   const navItems = LOCAL_NAV.filter((navItem) => {
     return (navItem.path !== 'subsidiaries' || headquarterType === 'ghq') &&
         (navItem.path !== 'timeline' || companyNumber) &&
-        (navItem.path !== 'advisers' || res.locals.features['companies-advisers'])
```
Comment (yeahfro) [inline src/apps/companies/middleware/local-navigation.js]: This should be a separate commit with corresponding unit test changes

File: src/apps/companies/transformers/one-list-core-team-to-collection.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+const isAccountManager = adviser => adviser.is_global_account_manager
+const isAdviser = adviser => !adviser.is_global_account_manager
+
+module.exports = function transformOneListCoreTeamToCollection (advisers) {
```
Comment (yeahfro) [inline src/apps/companies/transformers/one-list-core-team-to-collection.js]: The unit tests need updating for this function

**Comment by marcofucci** [comment]
Looks nice! Just 2 small comments:
1. It looks like the columns in the two tables are not aligned, would be nice if they were
2. Advisor should be renamed to Adviser

File: src/apps/companies/transformers/one-list-core-team-to-collection.js [hunk 7]
```diff
@@ -1,25 +1,22 @@
...
-    for (let teamMember of team) {
-      const { adviser } = teamMember
-      const location = adviser.dit_team.uk_region.name.length > 0 ? adviser.dit_team.uk_region.name : adviser.dit_team.country
-      const record = {
-        name: `${adviser.first_name} ${adviser.last_name}`,
-        team: adviser.dit_team.name,
-        location,
-      }
-      records.push(record)
+  const flattenTeam = (teamMember) => {
```
Comment (yeahfro) [inline src/apps/companies/transformers/one-list-core-team-to-collection.js]: This is definitely an improvement.

What do you think of this..?

```javascript
const mapAdviser = ({ adviser }) => {
  const { name, dit_team: team } = adviser
  return {
    name,
    team,
    location: get(team, 'uk_region.name', team.country),
  }
}

Comment (yeahfro) [reply src/apps/companies/transformers/one-list-core-team-to-collection.js]: In addition I would expect this commit to be a fixup which is then squashed into the original commit.

File: src/apps/companies/transformers/one-list-core-team-to-collection.js [hunk 8]
```diff
@@ -1,25 +1,22 @@
 const isAccountManager = adviser => adviser.is_global_account_manager
 const isAdviser = adviser => !adviser.is_global_account_manager
```
Comment (yeahfro) [inline src/apps/companies/transformers/one-list-core-team-to-collection.js]: Line break?

File: src/apps/companies/controllers/advisers.js [hunk 9]
```diff
@@ -11,14 +11,19 @@ async function renderAdvisers (req, res, next) {
   try {
     const { name: companyName, id: companyId } = res.locals.company
     const token = req.session.token
-    const coreTeam = await getCoreTeam(token, companyId)
+    const columns = {
```
Comment (yeahfro) [inline src/apps/companies/controllers/advisers.js]: These should go in `labels.js`. See `clientRelationshipManagementLabels` for a similar pattern.

File: src/apps/companies/controllers/advisers.js [hunk 10]
```diff
@@ -1,5 +1,5 @@
 const { notFound } = require('../../../middleware/errors')
-const { getCoreTeam } = require('../repos')
+const { getOneListGroupCoreTeam } = require('../repos')
```
Comment (yeahfro) [inline src/apps/companies/controllers/advisers.js]: `getCoreTeam` can be made redundant

File: test/unit/data/companies/core-team.json [hunk 11]
```diff
@@ -19,5 +19,26 @@
       "id": "8eefe6b4-2816-4e47-94b5-a13409dcef70"
```
Comment (yeahfro) [inline test/unit/data/companies/core-team.json]: This JSON file can be made redundant in favour of `one-list-group-core-team.json`

File: src/apps/companies/middleware/local-navigation.js [hunk 12]
```diff
@@ -10,11 +10,12 @@ function setCompaniesLocalNav (req, res, next) {
...
+        (navItem.path !== 'advisers' || (res.locals.features['companies-advisers'] && isOneListGroupTier))
```
Comment (yeahfro) [inline src/apps/companies/middleware/local-navigation.js]: This should also be a separate commit

Comment (yeahfro) [reply src/apps/companies/middleware/local-navigation.js]: And a corresponding unit test is required

**Comment by yeahfro** [comment]
If you want me to help you with the commits on this then let me know.

File: .gitignore (hunk 1 above)
Comment (retex73) [reply .gitignore]: @yeahfro I agree. I'll take care of this in the next PR

File: src/apps/companies/views/_advisers-table.njk (hunk 2 above)
Comment (retex73) [reply src/apps/companies/views/_advisers-table.njk]: @yeahfro I have changed this to use the existing data-table macro and extended its attributes to make it an equal width table

File: src/apps/companies/controllers/advisers.js (hunk 4 above)
Comment (retex73) [reply src/apps/companies/controllers/advisers.js]: @yeahfro Changed this now to use getOneListGroupCoreTeam

File: src/apps/companies/transformers/one-list-core-team-to-collection.js (hunk 6 above)
Comment (retex73) [reply src/apps/companies/transformers/one-list-core-team-to-collection.js]: @yeahfro The unit test (core-team-to-collection.test.js) now reflects this

File: src/apps/companies/transformers/one-list-core-team-to-collection.js (hunk 7 above)
Comment (retex73) [reply src/apps/companies/transformers/one-list-core-team-to-collection.js]: @yeahfro That's a nice improvement. Implemented!

File: src/apps/companies/controllers/advisers.js (hunk 9 above)
Comment (retex73) [reply src/apps/companies/controllers/advisers.js]: @yeahfro I created a new section in labels.js for this and implemented

File: src/apps/companies/controllers/advisers.js (hunk 10 above)
Comment (retex73) [reply src/apps/companies/controllers/advisers.js]: @yeahfro Removed!

File: src/apps/companies/middleware/local-navigation.js (hunk 12 above)
Comment (retex73) [reply src/apps/companies/middleware/local-navigation.js]: @yeahfro Made the changes to #14934773

File: src/apps/companies/transformers/one-list-core-team-to-collection.js (hunk 3 above)
Comment (retex73) [reply src/apps/companies/transformers/one-list-core-team-to-collection.js]: @yeahfro Implemented!

File: src/apps/companies/middleware/local-navigation.js (hunk 5 above)
Comment (retex73) [reply src/apps/companies/middleware/local-navigation.js]: @yeahfro done!

File: src/apps/companies/transformers/one-list-core-team-to-collection.js (hunk 8 above)
Comment (retex73) [reply src/apps/companies/transformers/one-list-core-team-to-collection.js]: @yeahfro line break added

**Comment by retex73** [comment]
@yeahfro Cheers for the input. Please would you review

File: src/apps/companies/controllers/advisers.js [hunk 13]
```diff
@@ -10,16 +12,29 @@ async function renderAdvisers (req, res, next) {
   try {
     const { name: companyName, id: companyId } = res.locals.company
     const token = req.session.token
-
-    const coreTeam = await getCoreTeam(token, companyId)
+    const { global_account_manager: globalAccountManager, adviser_on_core_team: adviserOnCoreTeam, location, team } = coreTeamLabels
+    const columns = {
+      'global_account_manager': {
```
Comment (yeahfro) [inline src/apps/companies/controllers/advisers.js]: These fields do not need `'`

File: test/unit/apps/companies/middleware/local-navigation.test.js [hunk 14]
```diff
@@ -131,9 +131,11 @@ describe('Companies local navigation', () => {
     })
   })
 
-  context('when the company advisers feature is enabled', () => {
+  context('when the company advisers feature is enabled and is One List Group Tier', () => {
     beforeEach(() => {
-      this.res.locals.company = {}
+      this.res.locals.company = {
+        one_list_group_tier: true,
```
Comment (yeahfro) [inline test/unit/apps/companies/middleware/local-navigation.test.js]: We usually make the data more realistic for testing
